src/components/panel/panel.tsx

```diff
@@ -109,30 +101,36 @@ export const EuiPanel: FunctionComponent<EuiPanelProps> = ({
   borderRadius = 'm',
   color = 'plain',
   hasShadow = true,
-  hasBorder,
+  hasBorder = false,
   grow = true,
   panelRef,
   element,
   ...rest
 }) => {
+  const euiTheme = useEuiTheme();
   // Shadows are only allowed when there's a white background (plain)
-  const canHaveShadow = color === 'plain';
+  const canHaveShadow = !hasBorder && color === 'plain';
   const canHaveBorder = color === 'plain' || color === 'transparent';
 
+  const styles = euiPanelStyles(euiTheme);
+  const cssStyles = [
+    styles.euiPanel,
+    grow && styles.grow,
+    styles.radius[borderRadius],
+    useEuiPaddingCSS()[paddingSize],
+    useEuiBackgroundColorCSS()[color],
+    canHaveShadow && hasShadow === true && styles.hasShadow,
+    canHaveBorder && hasBorder === true && styles.hasBorder,
+    rest.onClick && styles.isClickable,
```

Comment:
super optional comment - I find the `&&` syntax hard to discern in terms of what is actually getting fed into Emotion and would slightly prefer a ternary for explicitness (is it `onClick` or is it `isClickable`?). If you don't though, no worries 🤷 

```suggestion
    canHaveShadow && hasShadow === true ? styles.hasShadow : undefined,
    canHaveBorder && hasBorder === true ? styles.hasBorder : undefined,
    rest.onClick ? styles.isClickable : undefined,
```

Reply:
This is the pattern we've been using in all the conversion PR's and I personally do find the simplicity of something like 

```
prop.exists && styles[prop]
```

to be much simpler to read than the ternary at least when it comes to skimming and seeing what are all the `styles` keys pulling in because you can just read down the right hand side.

I can understand the more complex version like the border and shadow ones being harder to grok because there's an extra check in there. I could at least pull those out to evaluate to a constant first in order to get it down to one check if that's helpful.

Reply:
honestly I'm not super fussed either way in the grand scheme of things, just leaving my 2c on what's easier for me when I'm skimming. def down to personal preference and I'm totally fine leaving this as-is